packages/docusaurus-theme-classic/src/theme/Admonition/Types.tsx

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import Translate from '@docusaurus/Translate';
+import type {Props} from '@theme/Admonition';
+import IconNote from '@theme/Admonition/Icon/Note';
+import IconTip from '@theme/Admonition/Icon/Tip';
+import IconInfo from '@theme/Admonition/Icon/Info';
+import IconCaution from '@theme/Admonition/Icon/Caution';
+import IconDanger from '@theme/Admonition/Icon/Danger';
+import type {AdmonitionTypeConfig} from '@theme/Admonition/Types';
+
+function getInfimaClassName(suffix: string) {
+  return `alert alert--${suffix}`;
+}
+
+// eslint-disable-next-line @typescript-eslint/consistent-indexed-object-style
+const admonitionTypes: Record<Props['type'], AdmonitionTypeConfig> = {
```

Comment:
@Josh-Cena I'm thinking adopting another strategy for this map of admonitions components

I don't believe it's fair that we ask custom admonitions to respect our default `keyword/label/icon/className` structure: we should offer more freedom

Maybe just having a `keyword: Component` mapping would be better? users could implement admonitions without title/icon for example if they want to.

For example:

```jsx
const types = {
  note: NoteAdmonition
}

export default function NoteAdmonition({ keyword, title, children }) {
  return (
    <AdmonitionLayout title={title} icon={<NoteIcon />}>
      {children}
    </AdmonitionLayout>
  );
}

```

Does it make sense?

Reply:
implemented